abci/example/kvstore/kvstore.go

```diff
@@ -494,8 +494,13 @@ func (app *Application) getValidators() (validators []types.ValidatorUpdate) {
 	}
 	for ; itr.Valid(); itr.Next() {
 		if isValidatorTx(itr.Key()) {
-			validator := new(types.ValidatorUpdate)
-			err := types.ReadMessage(bytes.NewBuffer(itr.Value()), validator)
+			var (
+				validator       = new(types.ValidatorUpdate)
+				value           = itr.Value()
+				valueWorkingCpy = make([]byte, len(value))
+			)
+			copy(valueWorkingCpy, value)
+			err := types.ReadMessage(bytes.NewBuffer(valueWorkingCpy), validator)
```

Comment:
Why do we need a copy here? We're unmarshaling value using Protobuf.

Reply:
You are right, it's unnecessary to copy here. I added the copy thinking about the `bytes.NewBuffer` call, whose docs say `The new [Buffer] takes ownership of buf, and the caller should not use buf after this call`. But going down the rabbit hole of calls from `types.ReadMessage` I see we only read the value.
I will revert this commit, thanks for catching this! 😄